replica_aggregator/src/main/scala/uk/ac/wellcome/platform/storage/replica_aggregator/services/ReplicaAggregatorWorker.scala

```diff
@@ -35,26 +35,24 @@ class ReplicaAggregatorWorker[IngestDestination, OutgoingDestination](
 
   override def processMessage(
     payload: EnrichedBagInformationPayload
-  ): Try[IngestStepResult[ReplicationAggregationSummary]] = {
-    val replicaResult = ReplicaResult(payload)
+  ): Try[IngestStepResult[ReplicationAggregationSummary]] =
+    for {
 
-    val replicationSet = ReplicationSet(
-      path = ReplicaPath(payload.bagRootLocation.path),
-      results = Set(replicaResult)
-    )
+      aggregation <- replicaAggregator.aggregate(ReplicaResult(payload))
 
-    val summary = ReplicationAggregationComplete(
-      replicationSet = replicationSet,
-      startTime = Instant.now,
-      endTime = Instant.now
-    )
+      // TODO: Need to distinguish result to determine outgoing message:
+      // ReplicationAggregationIncomplete/ReplicationAggregationComplete
+      // How does that map to IngestStepSucceeded & then sendIfSuccessful?
 
-    val ingestStep = IngestStepSucceeded(summary)
+      ingestStep <- Success(aggregation match {
```

Comment:
Do you need to wrap this in a `Success` and a map? Could you do:

```scala
ingestStep = aggregation match {
   ...
}
```

Reply:
I think so. `IngestFailed` & `IngestStepSucceeded` are both rightly `Success` for the `processMessage` method.